File: WomenSafetyApp/android/app/src/main/java/com/anonymous/WomenSafetyApp/ShakeServiceModule.java
package com.anonymous.WomenSafetyApp;

import android.content.Intent;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;

public class ShakeServiceModule extends ReactContextBaseJavaModule {
    private final ReactApplicationContext reactContext;

    public ShakeServiceModule(ReactApplicationContext context) {
        super(context);
        this.reactContext = context;
    }

    @Override
    public String getName() {
        return "ShakeServiceModule";
    }

    @ReactMethod
    public void startService() {
        Intent serviceIntent = new Intent(reactContext, ShakeService.class);
        reactContext.startForegroundService(serviceIntent);
    }

    @ReactMethod
    public void stopService() {
        Intent serviceIntent = new Intent(reactContext, ShakeService.class);
        reactContext.stopService(serviceIntent);
    }

    @ReactMethod
    public void setEmergencyContact(String phoneNumber, String message, Promise promise) {
        EmergencyContactHelper.setContact(phoneNumber, message);
        promise.resolve("Contact set: " + phoneNumber);
    }
}

File: WomenSafetyApp/android/app/src/main/java/com/anonymous/WomenSafetyApp/ShakeServiceModule.kt
package com.anonymous.WomenSafetyApp

import android.content.Intent
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.module.annotations.ReactModule

@ReactModule(name = ShakeServiceModule.NAME)
class ShakeServiceModule(reactContext: ReactApplicationContext) :
    ReactContextBaseJavaModule(reactContext) {

    companion object {
        const val NAME = "ShakeServiceModule"
    }

    override fun getName(): String = NAME

    @ReactMethod
    fun startService() {
        val intent = Intent(reactApplicationContext, ShakeService::class.java)
        reactApplicationContext.startService(intent)
    }

    @ReactMethod
    fun stopService() {
        val intent = Intent(reactApplicationContext, ShakeService::class.java)
        reactApplicationContext.stopService(intent)
    }

    @ReactMethod
    fun setEmergencyContact(number: String, message: String) {
        // store contact in SharedPreferences or handle as needed
    }
}
